src/decrypters/mediafoundation/MFDecrypter.cpp

```diff
@@ -0,0 +1,173 @@
+#include "MFDecrypter.h"
+#include "MFCencSingleSampleDecrypter.h"
+#include "../../utils/Base64Utils.h"
+#include "../../utils/log.h"
+
+#include <mfcdm/MediaFoundationCdm.h>
+#include <mfcdm/MediaFoundationCdmSession.h>
+#include <mfcdm/MediaFoundationCdmModule.h>
+#include <mfcdm/MediaFoundationCdmFactory.h>
+#include <mfcdm/MFException.h>
+
+#include <kodi/Filesystem.h>
+
+using namespace UTILS;
+using namespace DRM;
+using namespace kodi::tools;
+
+CMFDecrypter::~CMFDecrypter()
+{
+  delete m_cdm;
+}
+
+const char* CMFDecrypter::SelectKeySytem(const char* keySystem)
+{
+  if (strcmp(keySystem, "com.microsoft.playready"))
+    return nullptr;
+  return "urn:uuid:9A04F079-9840-4286-AB92-E65BE0885F95";
+}
+
+bool CMFDecrypter::OpenDRMSystem(const char* licenseURL,
+                                 const AP4_DataBuffer& serverCertificate,
+                                 const uint8_t config)
+{
+  m_cdm = new MediaFoundationCdm();
+
+  if (!(config & DRM::IDecrypter::CONFIG_PERSISTENTSTORAGE))
+  {
+    LOG::Log(LOGERROR, "MF PlayReady requires persistent storage.");
+    return false;
+  }
```

Comment:
maybe im wrong but this condition seem weird, for widevine this is not mandatory
why should be on playready? take in account that we dont allow to store encrypted content to device storage for later uses nor store license